src/lib/process/index.ts

```diff
@@ -68,11 +72,28 @@ export const createProcess = async (
       program,
     }
 
-  const process: Process.Payload = await createProcessTransaction(polkadot, processId, program, options)
+  if (version === currentVersion) {
+    const process = await getProcess(polkadot, processId, version)
+
+    if (program.length !== process.program.length) return {
+      message: 'existing: programs are different lengths',
+      process,
+    }
+
+    if (!program.every((step, i) => textify(step) === textify(process.program[i]))) return {
+      message: 'existing: program steps did not match',
+      process,
```

Comment:
these two are `throw new someError` cases?

Reply:
as in we want to communicate to the user that they've got something wrong

Reply:
No, it does not throw an error

this is what we do in regards to communications also returning the old/current/existing process.

Reply:
my point is consistency on the structure of what we return with a 'bad input' case. A bad input case being e.g. putting the wrong version, invalid program, not matching existing if the version number is the same

Reply:
inconcistency - here we are returning `message, process` but with a version error we're throwing an error

Reply:
This implementation does address multiple process -> l3-68:
```
When multiple processes are given to create if some of them pass and some fail, the output only throws an error for the failed one. It should give a success/fail message for each process
```
- this is the main reason, happy to make code more complex by try/catch and address there, in the **loadProcess** https://github.com/digicatapult/dscp-process-management/blob/3ae687676a0e88c3a1eb5a8ea9a6cd2efe7967a0/src/lib/process/index.ts#L26 method so we can throw error for single and skip for multiple

In case you try to create a new process on it's own it would return the error message along with current/old process. To me it feels like it does cover the AC.

As I said happy to change, but then we will have not less consistency, because for multiple we skip for single we throw?

Reply:
The reasoning for the AC on this is pretty unclear to me so I think we can be flexible and then do another PR if we find this behaviour is unhelpful. In other words for now this looks fine

Reply:
I think has been resloved now. Please confirm